pkg/cli/cli.go

```diff
@@ -47,14 +52,61 @@ func ParseConfig(configFile string) (vsphere.Config, error) {
 	return cfg, err
 }
 
-// TODO (fanz) : Perform vSphere configuration health check on VM:
-func CheckVSphereConfig(config vsphere.Config) error {
+// CheckVSphereConfig performs vSphere health check on VMs
+// TODO (fanz) : support checking network
+func CheckVSphereConfig(ctx context.Context, o *ClientOption) error {
+	c, err := o.GetClient()
+	if err != nil {
+		return err
+	}
+	vc := view.NewManager(c)
+	cv, err := vc.CreateContainerView(ctx, c.ServiceContent.RootFolder, []string{"VirtualMachine"}, true)
+	if err != nil {
+		return err
+	}
+	defer cv.Destroy(ctx)
+	var vms []mo.VirtualMachine
+	var vm *object.VirtualMachine
+	var config extraConfig
+	config.Set("disk.enableUUID=1")
+
+	f := find.NewFinder(c, true)
+	err = cv.Retrieve(ctx, []string{"VirtualMachine"}, []string{"summary"}, &vms)
+	if err != nil {
+		return err
+	}
+	for _, v := range vms {
+		if v.Summary.Config.Uuid == "" {
+			name := v.Summary.Config.Name
+			vm, err = f.VirtualMachine(ctx, name)
+			if err != nil {
+				return err
+			}
+
+			spec := vimType.VirtualMachineConfigSpec{
+				ExtraConfig: config,
+			}
+			task, err := vm.Reconfigure(ctx, spec)
```

Comment:
With the ContainerView selecting all VMs in the inventory, this means we're Reconfiguring every VM when we should only do this for k8s nodes.  I believe the `NodeManager` related code can help to find the k8s VMs, @dvonthenen may have pointers.

Reply:
I have a first look at the `NodeManager` code. Basically all methods are not exposed to other packages, it's seems like difficult to leverage code from NodeManager. I will look into it to see what's the better solution. Please feel free to share any ideas. @dvonthenen @dougm .
Thanks so much!

Reply:
We can refactor the NodeManager code and export the functionality needed, probably into a package that can be shared by both.

Reply:
One concern is (might not be critical for this case) that keeping multi-connection to vc by different clients (`NodeManager` sets up a `vSphereConnection`, and vcpctl uses its own `vim25.Client`) here could cause performance issue or even race condition on vm operations.

Reply:
How about for now, change the PR to check if the property `disk.enableUUID == 1` and log a message if it isn't? The message could even include the govc command to set the property :)

We should check if the property is already set to avoid unneeded `Reconfigure` anyhow.  We can add the node check and `Reconfigure` in a later PR.

Reply:
@dvonthenen I am thinking if `NodeManager` can expose a method to return a nodes list (just name is fine for now). By this kind of method, vcpctl can perform a node check for only `Reconfigure` these particular k8s nodes.

Reply:
Separate PR sounds good to me.

Reply:
@fanzhangio @dougm yea, that shouldn't be a problem and it's fairly easy to do. if we do expose a call, we should go through the process of making this the start of a formal API that the provider exposes. I can start working on that on Monday.